lms/djangoapps/discussion/api/external_discussions.py

```diff
@@ -0,0 +1,44 @@
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from .data import ExternalDiscussionData
+from ..models import ExternalDiscussionsIdMapping
+
+
+def get_external_discussion_context(context_key: CourseKey, usage_key: UsageKey) -> ExternalDiscussionData:
+    try:
+        discussion_mapping = ExternalDiscussionsIdMapping.objects \
+            .get(context_key=context_key, usage_key=usage_key)
+    except ExternalDiscussionsIdMapping.DoesNotExist:
+        # Could happen if it hasn't been published.
+        raise ExternalDiscussionData.DoesNotExist(
+            "No ExternalDiscussionData for context_key={context_key} " \
+            "and usage_key={usage_key}".format(
+                context_key=context_key,
+                usage_key=usage_key
+            )
+        )
+
+    return ExternalDiscussionData(
+        context_key=discussion_mapping.context_key,
+        external_discussion_id=discussion_mapping.external_discussion_id,
+        usage_key=discussion_mapping.usage_key
+    )
+
+
+def get_internal_discussion_context(external_discussion_id: str) -> ExternalDiscussionData:
+    try:
+        discussion_mapping = ExternalDiscussionsIdMapping.objects \
+            .get(external_discussion_id=external_discussion_id)
+    except ExternalDiscussionsIdMapping.DoesNotExist:
+        # Could happen if it hasn't been published.
+        raise ExternalDiscussionData.DoesNotExist(
+            "No ExternalDiscussionData for external_discussion_id={external_discussion_id}".format(
+                external_discussion_id=external_discussion_id
+            )
+        )
```

Comment:
Instead of raising an exception, perhaps we can just return None? 
Not having a external discussion id doesn't seem like an exceptional case.

Reply:
Done in next commit